.github/workflows/pre-merge.yaml

```
@@ -101,4 +101,4 @@ jobs:
       - name: Run with allWarningsAsErrors
         uses: gradle/gradle-build-action@3fbe033aaae657f011f88f29be9e65ed26bd29ef # tag=v2
         with:
-          arguments: build -x detekt -PwarningsAsErrors=true
+          arguments: compileKotlin compileTestKotlin compileTestFixturesKotlin -PwarningsAsErrors=true
```

I would go with `classes testClasses jar`, because it's more stable in the long run, it would contain anything that is required for compilation, kapt, ksp, plugin generation etc.
```suggestion
          arguments: classes testClasses jar -PwarningsAsErrors=true
```

---

\+ `functionalTestClasses`

```suggestion
          arguments: classes testClasses functionalTestClasses jar -PwarningsAsErrors=true
```

---

Doesn't that just compile more things that I don't care? `warningsAsErrors=true` is a flag that only the kotlin compiler reads.

---

KotlinCompile is not a final class, other tasks might extend it. Calling the `compile*Kotlin` tasks directly relies on implementation details of KGP. `classes` / `jar` are documented Gradle lifecycle methods.

---

> Calling the compile*Kotlin tasks directly relies on implementation details of KGP

+1 on this

---

My 2c is that `compileKotlin compileTestKotlin compileTestFixturesKotlin` captures exactly what we need. That's detekt code that we have control over that we need to warn about.

> Calling the compile*Kotlin tasks directly relies on implementation details of KGP

This isn't true. [Here](https://kotlinlang.org/docs/gradle-compiler-options.html#how-to-define-options) it explicitly says that the tasks can and should be referenced directly by name - "When targeting the JVM, the tasks are called compileKotlin for production code and compileTestKotlin for test code. The tasks for custom source sets are named according to their compile<Name>Kotlin patterns."

> classes / jar are documented Gradle lifecycle methods.

Just FYI `jar` isn't a lifecycle task (lifecycle tasks are tasks that do not do work themselves, and `jar` task does some work as it assembles the JAR archive).

---

Oh, nice RTFM, thanks. True about jar!

I guess if this is the case we can go with explicit refs. What will prevent though future other warnings from showing up if there are more tasks/source sets added?
Someone noticing locally and raising a PR? (I guess that's acceptable, that's what most projects do, except the raising a PR part 😞.)